documentation/design/proposals/management_tasks-for-shared.adoc

```diff
@@ -0,0 +1,88 @@
+messages (HTTP and AMQP) to action the managemnet requests and relies on the existance of the `agent` component (something
+that will disappear with shared infrastructure.
+
+We need new design for this part of the system in order to deliver shared.
+
+== Current Design Flaws
+
+The current design has number of drawbacks:
+
+1. The design assumes that the management request can be completed within a HTTP request timeout (of both the use-agent
+   and HTTP proxies).  If the management request takes too long the response channel is lost meaning the user has no
+   mechanism to monitor the request's completion.  Purging a deep queue or moving a significant number of messages
+   is likely to breach timeouts. 
+1. The deisgn has no mechnaism to monitor the progress of a request
+1. The deisgn has no mechnaism to cancel a management request.
+1. The Console UI uses modals UI elements to initiate the request and blocks awaiting the response. The user is unable
+   to continue to interact with the app whilst the request is in progress.
+1. There is no provision for a command line interface.
+1. Poorly defined RBAC model.
+
+== High Level Proposal
+
+Introduce a new CRD `ManagementTask`.  A `ManagementTask` executes a management function (such as a purge) in a
+tenant namespace within EnMasse.  It would be rather similar to
+https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/[Jobs] from Kubernetes.
+
+* The Tenant Admin would create an instance of the resource in order to cause the task to occur.  The contents of the
+  resource describe the jobs and its parameters.
+* The EnMasse Operator actions the tasks - initiating requests to Brokers or Routers as necessary.  This might invoke
+  the invocation of Router/Broker management functions (close connection, clear queue etc), or might involve putting down
+  router wiring to implement a move message task.
+* The EnMasse Operator periodically updates the task with progress of the task thus allowing the Tenant Admin to monitor
+  the task for completion.  The availablility of progress information depends on the type of task.
+* The Tenant Admin may delete a ManagementTask, this would be a signal to the Operator to cancel the task (if possible).
+* The Operator would mark the ManagementTask as compete. A `ttlSecondsAfterFinished` parameter would allow the Operator to 
+  clean up the ManagementTasks automatically.
+
+The tenant admin would be able to CRUDWL `ManagementTask` from the CLI using `kubectl`/`oc` tooling (usual RBAC permissions
+apply), or later an EnMasse API.
+
+From the Console. functions such as purge queue/subscription would, behind the button cause an instance of this object to
+be created.  There would be an area of the UI which lists running/recently finsihed tasks.  From this page, you'd be able
+to cancel the task too.
+
+=== CRD sketch
+
+```
+apiVersion: enmasse.io/v1
+kind: ManagementTask
+metadata:
+  name: mytenantnamespace
```

Comment:
name is arbitrary - the tenant would create this resource in his messaging project.

Reply:
Done.